schemas/iso19115-3.2018/src/main/plugin/iso19115-3.2018/convert/fromJsonOpenDataSoft.xsl

```diff
@@ -244,9 +264,16 @@
               </mri:pointOfContact>
             </xsl:for-each>
 
-            <mri:topicCategory>
-              <mri:MD_TopicCategoryCode></mri:MD_TopicCategoryCode>
-            </mri:topicCategory>
+            <!-- ODS themes copied as topicCategory -->
```

Comment:
Not sure if `metas/theme` correspond to what ODS calls "dataset theme" ? But if you read the ODS documentation about "dataset themes" (see https://userguide.opendatasoft.com/l/en/article/7ng5ysoinv-managing-the-dataset-themes), a theme is a list of free text terms with 12 default values provided that any ODS instances can customize.

So it does not sounds appropriate to map this to an ISO topic category which is an enumeration (see https://github.com/geonetwork/core-geonetwork/blob/main/schemas/iso19115-3.2018/src/main/plugin/iso19115-3.2018/schema/standards.iso.org/19115/-3/mri/1.0/identification.xsd#L356-L463) (ie. a list of values that you cannot extend unless you create a profile). 

Putting free text values into an enumeration will brings a couple of issues:
* All records produced by that harvester will be invalid

![Pasted image 1](https://github.com/geonetwork/core-geonetwork/assets/1701393/e21cef57-1db2-4a46-9294-68c64920c162)

* Facets will be a mix of keys and free text values - possibly in a mix of languages ... even those with same meaning will be messed up eg. Santé = Health ?

![Pasted image](https://github.com/geonetwork/core-geonetwork/assets/1701393/3abe4212-8f61-4d3d-9e5b-aaf3c4c5231c)


* Topic category icons will not be available for the free text values

If you really want to put ODS dataset theme in ISO topic category, you should provide a mapping and set the proper ISO keys. Therefor as it is more something like category or thesaurus in GeoNetwork, I would rather revert the change and keep ODS dataset themes in a dedicated keyword block so that user can easily create facet or search on this ODS vocabulary ...

Reply:
The rationale was that records harvested from ODS would be filterable by theme just like any other one. The record will obviously be invalid but the search experience is better. Since themes in ODS are pretty much free form it's probably impossible to map them to an ISO topic.

Feel free to revert this change if you think it's inappropriate.

Reply:
Why not simply map the default ODS themes to valid ISO topics eg. `metas/theme=Santé` then `<mri:MD_TopicCategoryCode>health` and keep custom and unknown ODS themes in a dedicated keyword group. This would allow to conform to ISO and not mess up the GeoNetwork default filter on ISO topics.

If building new UI with a concept of theme which is not corresponding to ISO topics, then the ODS keyword group can be used and/or combined with ISO topics for filtering.

Other alternatives:
* Other harvesters not providing such information have extra parameters to set language or topic category eg. OGC WxS

![image](https://github.com/geonetwork/core-geonetwork/assets/1701393/d8a3bb3c-eea2-4781-97ac-0c1bc59bd5e2)

* Batch editing can also be used to extend harvested records with local rules (eg. to add keywords from an internal classification system).